src/Model/Behavior/FootprintBehavior.php

```diff
@@ -80,7 +81,18 @@ public function beforeFind(Event $event, Query $query, ArrayObject $options)
 
         foreach (array_keys($config) as $field) {
             $path = $this->config('propertiesMap.' . $field);
-            if ($value = Hash::get((array)$options, $path)) {
+
+            $check = false;
+            $query->traverseExpressions(function ($expression) use (&$check, $field, $query) {
+                if ($expression instanceof IdentifierExpression) {
+                    !$check && $check = $expression->getIdentifier() === $field;
+                    return;
+                }
+                $alias = $this->_table->aliasField($field);
+                !$check && $check = preg_match('/^' . $alias . '/', $expression->sql($query->valueBinder()));
```

Comment:
Nope, that only covered conditions added by the behavior not user added conditions. We'll see about that later, merging.